File: app/src/main/java/com/moon/mytools/MainActivity.java
package com.moon.mytools;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

import com.alibaba.android.arouter.facade.annotation.Autowired;
import com.alibaba.android.arouter.facade.annotation.Route;
import com.alibaba.android.arouter.launcher.ARouter;
import com.moon.common.ui.component.MBaseActivity;
import com.moon.mlibrary.log.MLog;
import com.moon.mlibrary.util.ActivityManager;
import com.moon.mytools.logic.MainActivityLogic;

/**
 * Date: 9/30/21 2:29 PM
 * Author: Moon
 * Desc: 主页
 */
@Route(path = "/index/main")
public class MainActivity extends MBaseActivity implements MainActivityLogic.ActivityProvider, ActivityManager.FrontBackCallback {

    @Autowired
    public int age;

    private MainActivityLogic mMainActivityLogic;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        ARouter.getInstance().inject(this);

        mMainActivityLogic = new MainActivityLogic(this, savedInstanceState);

        ActivityManager.getInstance().registerFrontBackCallback(this);

        FragmentManager fragmentManager = getSupportFragmentManager();
        FragmentTransaction transaction = fragmentManager.beginTransaction();

        transaction.commit();
    }

    @Override
    protected void onSaveInstanceState(@NonNull Bundle outState) {
        super.onSaveInstanceState(outState);
        mMainActivityLogic.onSaveInstanceState(outState);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        ActivityManager.getInstance().removeFrontBackCallback(this);
    }

    @Override
    public void onChanged(boolean front) {
        MLog.d("app处于：" + front);
    }
}

File: app/src/main/java/com/moon/mytools/MainActivity.kt
package com.moon.mytools

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.moon.mlibrary.log.*
import com.moon.mytools.demo.VideoListActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    private var mViewPrinter: MViewPrinter? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btMovie.setOnClickListener {
            startActivity(Intent(this, VideoListActivity::class.java))
        }

        MLog.d(Person("lx", 30))

        mViewPrinter = MViewPrinter(this)
        mViewPrinter!!.viewProvider.showFloatingView()

        showLog.setOnClickListener {
            MLogManager.getInstance().addPrinter(mViewPrinter)
            MLog.log(
                object : MLogConfig() {
                    override fun includeThread(): Boolean {
                        return true
                    }
                },
                MLogType.E,
                "---",
                "55555"
            )
        }
    }

    class Person(var name: String, var age: Int)
}
